### ISSUE TITLE
gps-entity-place attribute not working in Southern Hemisphere

### ISSUE THREAD

**Issue by maletiger0404** [AR-js-org/AR.js#265]
https://user-images.githubusercontent.com/67555770/117088916-3a04eb80-ad97-11eb-9132-14d89a43628e.mp4

Why the image is displayed on the floor?
I don't know what is wrong with my code.

![image](https://user-images.githubusercontent.com/67555770/117089013-91a35700-ad97-11eb-9c77-db0b9252f688.png)

If I remove the look-at attribute, I can not see anything.
I set the height above sea level for the y of the position.

the aframe.min.js and aframe-look-at-component.min.js versions are mismatched?

**Comment by nickw1** [comment]
@WebandMobileMaster see my post in the other issue.
It doesn't matter about the version numbers as they are independent projects.

**Comment by WebandMobileMaster** [comment]
https://user-images.githubusercontent.com/67555770/117220175-0c2caf00-ae4a-11eb-94f2-7a00343946de.mp4

@nickw1 I tried your sample code and I saw a big red rectangle on the street, but then when I tried again in my house I also saw a big rectangle so I think the location-based stuff is not working.

**Comment by maletiger0404** [comment]
I have added the look-at attribute but still the rectangle showing inside my house.

**Comment by maletiger0404** [comment]
In Southern Hemisphere of the earth, the Ar.js is not working?

**Comment by nickw1** [comment]
@maletiger0404 if you walk down the street away from the red box, does it get smaller?
Is it always to the north of you?
There should also be a yellow box, can you see that?

**Comment by maletiger0404** [comment]
No, I can not see the yellow box.
and If I walk down the street away from the red box, it isn't be smaller.

**Comment by maletiger0404** [comment]
Hi @nickw1 
Here is the news, I have tested it in India and it worked well.
But it should be worked in Australia. No needs other countries.
What do you think about this?

**Comment by maletiger0404** [comment]
https://user-images.githubusercontent.com/67555770/117289599-1f726580-aeb0-11eb-9701-de4be7b4dde8.mp4

When I move, why the images are moving together?

**Comment by maletiger0404** [comment]
Can you please solve the Southern Hemisphere issue and release new version?

**Comment by maletiger0404** [comment]
What is the way to solve the Southern Hemisphere issue?

**Comment by nicolocarpignoli** [comment]
@maletiger0404 
Hi, I need to clarify a point. This is open-source software. I get might be urgent for you, but Open Source is thought to be about authors and collaborator's self-commitment.

As this is probably an open bug, the things you can do, if you want this to be fixed NOW are:
- fix it by yourself
- pay someone to fix it for you.

You can talk with @nickw1, for example, as currently, I do not have time for this.
The "xs:code" platform of AR.js (for paid bug fixing) redirects to me but I have zero problems asking someone else to help with this paid fixing.

So basically I suggest you to not open several issues for the same problem, to not ask repeatedly for help and to offer a fee if someone has the time and the knowledge to help you.

**Comment by maletiger0404** [comment]
Okay, I see. But what are the differences in GPS of the Southern Hemisphere with Northern Hemisphere?

**Comment by nickw1** [comment]
@maletiger0404 in the Southern Hemisphere, latitude is negative. I would guess it is a logic error related to that. Unfortunately my own time is very tight during the month of May, while I *might* have the chance to look at it I cannot *guarantee* this at the present time. I am likely to have much more time in June. If you want to attempt it yourself, I would suggest looking at `gps-camera` and `gps-entity-place` within the `location-based` directory of `aframe`.

**Comment by maletiger0404** [comment]
Okay, I will try.

**Comment by maletiger0404** [comment]
https://user-images.githubusercontent.com/67555770/117520041-aaa34680-afe9-11eb-9d97-dd00a6d2de68.mp4


https://user-images.githubusercontent.com/67555770/117520051-b3941800-afe9-11eb-862e-01fb5f91e6c3.mp4

Hi, @nickw1 
Please check this video. The images are fixed on the phone.
What do you think about this? Can you give me any tips?

**Comment by maletiger0404** [comment]
![image](https://user-images.githubusercontent.com/67555770/117536360-f7b10800-b03d-11eb-8640-ef5ee7215fd6.png)
![image](https://user-images.githubusercontent.com/67555770/117536383-0f888c00-b03e-11eb-968f-4837de75cb6a.png)

Here is my code. Please give me any tips. If so, I will do it myself.

**Comment by nickw1** [comment]
@maletiger0404 another approach that might work for you, if you're happy checking out code from github: the pure three.js version of location-based AR, not yet integrated into the main AR.js repo yet, but will be merged in once the ES6 branch has been debugged and merged.

This code is more or less entirely independent of the A-Frame version so is unlikely to have the same bug, can't guarantee it but unlikely.

See:
https://github.com/nickw1/AR.js/tree/three-location-based/three.js/location-based

**Comment by nickw1** [comment]
@nicolocarpignoli unrelated, but I have just discovered a (long-standing) typo in `gps-entity-place`, namely the emitted event is misspelled as  `gps-entity-place-update-positon` (posit*on* not posit*ion). Nothing else in AR.js appears to use this, though that would explain why some code of mine was not working and I could not determine the cause. Shall I just fix this quickly on dev and master and re-release?

**Comment by nicolocarpignoli** [comment]
@nickw1 yes, please :)

**Comment by maletiger0404** [comment]
@nickw1  Did you fix the Southern Hemisphere issue?

**Comment by nickw1** [comment]
@maletiger0404 I am sorry to have to mention this again, but I have already mentioned twice that I may not have time to fix this. I had a quick look today but could not see anything obvious. I have directed you to an alternative version of AR.js which uses pure Three.js; do please look at this. It may well work for you.

**Comment by maletiger0404** [comment]
Thank you. I will try with the three.js

**Comment by maletiger0404** [comment]
> @maletiger0404 another approach that might work for you, if you're happy checking out code from github: the pure three.js version of location-based AR, not yet integrated into the main AR.js repo yet, but will be merged in once the ES6 branch has been debugged and merged.
> 
> This code is more or less entirely independent of the A-Frame version so is unlikely to have the same bug, can't guarantee it but unlikely.
> 
> See:
> https://github.com/nickw1/AR.js/tree/three-location-based/three.js/location-based

Hi, @nickw1 
Is there any document for this?

**Comment by maletiger0404** [comment]
![image](https://user-images.githubusercontent.com/67555770/117665449-2ff34a80-b1e6-11eb-9e25-6bfeac414778.png)


https://user-images.githubusercontent.com/67555770/117666341-1b638200-b1e7-11eb-8b04-a463af5b80ee.mp4


Hi @nickw1  I used the three.js and the cubes are rendered but as you can see on the video it is not rendered by the GPS position I think.
What do you think about this?

**Comment by nickw1** [comment]
@maletiger0404 can you try walking away from the cubes? It is difficult to see whether it is working or not unless you walk some distance (50 metres or thereabouts). Ideally post a video of this if you can.

**Comment by maletiger0404** [comment]
Hi, @nickw1 I walked away from the cubes but same issue.
Even if I go closer or away from the cubes, the distance between the cubes and my phone is fixed.

**Comment by nickw1** [comment]
@maletiger0404 are you sure your phone doesn't have some issue? For example, is it updating its location continuously? Can you get other people in Australia to test, as well as other people in the Northern Hemisphere? With both the A-Frame and the three.js versions? Only that way can we be sure it's definitely a Southern Hemisphere bug and not an issue with your phone.

If I have time I will try and test this 'virtually' on my laptop with fake Northern and Southern Hemisphere locations, with both the A-Frame and three versions. This may be some time though - as before, I make no guarantees whatsoever exactly when this will be done, my job is quite demanding at the moment.

The code you have circled in red has a typo `fotEach` rather than `forEach`.

**Comment by nickw1** [comment]
@maletiger0404 you might also find the link below helpful, it explains how to remote-debug apps running on Chrome in Android devices. For example this would have probably picked up the black-screen issue you mentioned above.
https://developer.chrome.com/docs/devtools/remote-debugging/

**Comment by maletiger0404** [comment]
Exactly, the GPS attribute is not working.

![image](https://user-images.githubusercontent.com/67555770/117919724-de0b0b80-b330-11eb-9d9c-78564c596934.png)

**Comment by nickw1** [comment]
It might possibly be to do with the `maximumAge` parameter to the GPS listener.

Would you be able to edit this file on the three version: 
https://github.com/nickw1/AR.js/blob/three-location-based/three.js/location-based/js/location-based.js
and remove the `maximumAge` parameter from line 27?

This would not be a Southern Hemisphere issue, but it might fix the problem; see what it does for you.

**Comment by maletiger0404** [comment]
I tested this in India and in Australia it's not working. So the Southern Hemisphere issue is not solved on this three.js version.
So now I am going to use the ARCore or Wikitude. giving up the ar.js
Do you think that it's the best way for me to implement the location-based AR?

**Comment by nickw1** [comment]
OK, sorry AR.js doesn't meet your needs. Have you definitely tested it with multiple devices in Australia though?
Could you give us your phone model, operating system (plus version) and browser (plus version)?
It may help us diagnose the issue.

Thanks.

**Comment by Aroustaei** [comment]
I don't think that the problem is related to the Southern Hemisphere. I'm in Northern Hemisphere and have the same issue, the AR content (glt model) is not dependant on the GPS location and will appear everywhere.

**Comment by maletiger0404** [comment]
yes, @Aroustaei  Ar.js has problems like that you mentioned in Northern Hemisphere but in Southern Hemisphere, the problem is serious than Northern.
It's even not displayed.

**Comment by jrmasiero** [comment]
Hi, I'm from Brazil. Tested on Porto Alegre (latitude: -30.051877; longitude: -51.183395) iphone 12 pro, ios 14.5.1, Safari 14.1)

https://youtu.be/_qO5DjVWS5c

Working fine.

**Comment by maletiger0404** [comment]
okay, it looks good but can you please test it again with walking on the street?

**Comment by nickw1** [comment]
@jrmasiero that is what I would expect - thanks for confirming it works ok in the Southern Hemisphere; the three.js code in particular is quite simple and I cannot see any reason why it would not work in the Southern Hemisphere.

@maletiger0404 could you let me know your phone/OS/browser plus versions as requested? 

@Aroustaei you also? There is obviously a bug of some sort, and I am wondering if it always happens on certain devices/browsers. Are you able to make your application public anywhere? Then I can test on my device.

@maletiger0404 could you try removing the `maximumAge` option as I mentioned above? I have the feeling that might fix it; it may be using a cached GPS position rather than the current one. Thanks.

**Comment by nickw1** [comment]
@Aroustaei sorry ignore the typo in my previous message, I meant to say "you also have an issue?" and forgot to complete the sentence.

**Comment by jrmasiero** [comment]
Walking test (objects scale "1 1 1") using aframe

https://youtu.be/xYeRRNfvPJ4

Code

https://glitch.com/edit/#!/ar-gps

**Comment by Aroustaei** [comment]
@nickw1  this is the code 
https://ribbon-garrulous-emu.glitch.me
and the model sticks to the camera when I'm walking. I'm using Android 6.0, HTC One M9PLUS Chrome 90.0.4430.91
Another problem I'm facing is that I can not change the scale of the model. I mean when I change the scale to for example "5 5 5" there is no difference.

**Comment by nickw1** [comment]
@Aroustaei ok thanks. That looks like the standard AR.js hello world example so it ought to work.
As I said I am wondering if it's to do with the `maximum-age` property of the GPS listener. Let me try something and when I have, I will give you an updated URL which you can test to see if that works.

**Comment by Aroustaei** [comment]
@nickw1 thank you so much... do you have any idea that why I can not make the AR content larger by Scale?

**Comment by nickw1** [comment]
@Aroustaei sorry.. not sure why that's not working for you, the `scale` attribute definitely does work: does it work if you replace your model with an `<a-box>`?

**Comment by maletiger0404** [comment]
Hi, everyone. If an object is rendered on the back of a building and if I stand on the front of the building, the object should not be shown, I think.
How can we solve this problem?

**Comment by Aroustaei** [comment]
@nickw1  I tried the scale attribute with the <a-box> and it works. it seems the problem with scaling is related to my own 3D model. So I'm working on it. If I don't want to show the texture, Is it necessary to have <.bin> file in addition to the <.gltf> file in the asset folder or the gltf file would be enough?

**Comment by maletiger0404** [comment]
> Walking test (objects scale "1 1 1") using aframe
> 
> https://youtu.be/xYeRRNfvPJ4
> 
> Code
> 
> https://glitch.com/edit/#!/ar-gps

Thank you so much.
Your code is working in Australia. I don't know well what is wrong with my code.
@nickw1 What ways are existing to increase the accuracy?
A box is displayed in the sky. How do I make only objects 5 meters away from the phone visible?

**Comment by maletiger0404** [comment]
How can I disable the screen draggable?

**Comment by nickw1** [comment]
@maletiger0404 @Aroustaei ok try this

https://hikar.org/265/

What this does is automatically create two boxes a short distance to your north as soon as it gets a GPS position (wherever in the world you are)

If you walk away from your original position, the boxes should get smaller, but if you walk north, they should get bigger.

This is using a version of AR.js with the `maximumAge` option turned off.

Does this work for you?

Thanks.

**Comment by Aroustaei** [comment]
thank you @nickw1 ....  for me when I changed my location the size of the boxes don't change and they are always in the same position.

**Comment by maletiger0404** [comment]
> @maletiger0404 @Aroustaei ok try this
> 
> https://hikar.org/265/
> 
> What this does is automatically create two boxes a short distance to your north as soon as it gets a GPS position (wherever in the world you are)
> 
> If you walk away from your original position, the boxes should get smaller, but if you walk north, they should get bigger.
> 
> This is using a version of AR.js with the `maximumAge` option turned off.
> 
> Does this work for you?
> 
> Thanks.

Okay, I will test and I want to hide the box if I walk from the box about 10 ~ 20 meters(I can set). not smaller.
I think we should calculate the distance between the camera and the box for this.

**Comment by nickw1** [comment]
@Aroustaei ok thanks. Strange.. it's as if the GPS position is not updating but I am not sure why.

@maletiger0404 you can use the `maxDistance` property of the `gps-camera` to specify a maximum distance at which objects are visible, in metres.

**Comment by nickw1** [comment]
@Aroustaei @maletiger0404 closing this issue, please use issue #275 for further discussion.

**Comment by maletiger0404** [comment]
Hi @jrmasiero  How are you?
Can you please test on your side?
Is it still working?

**Comment by nickw1** [comment]
To update, issue #275 has been closed. Please use #278 to report or discuss this issue.

**Comment by fahimaltinordu** [comment]
Hi @maletiger0404 did you solved the issue "Why the image is displayed on the floor?" ?
I have the same issue and cannot solve it

**Comment by nickw1** [comment]
@fahimaltinordu unfortunately we have not sourced the cause of this issue yet. Please see #278 and report your device, operating system, and browser (plus version). Thanks.

**Comment by maletiger0404** [comment]
> Hi @maletiger0404 did you solve the issue "Why the image is displayed on the floor?" ?
> I have the same issue and cannot solve it

No, I didn't. I think finding another module is better than fixing that issue. e.g. did you try the React Viro?

**Comment by fahimaltinordu** [comment]
@nickw1  @maletiger0404 
Now it (displaying image on the floor issue) works for me, i just changed script src link (aframe-ar-nft-js) with folder src. 

https://github.com/fahimaltinordu/AR-testz89thw/tree/gh-pages

However lat/long still not works, or i am making mistake with a-camera [simulateLatitude]. Although object doesnot come with me if i walk, it appears in any location

**Comment by nickw1** [comment]
Hi @fahimaltinordu does this example work for you?

https://hikar.org/265/

Once it gets a GPS location, it should place two red boxes perhaps 20 metres north of your current position. You will then be able to move towards them. Best tested on a mobile device.

If that works the problem is in your code, if not the problem is with AR.js.

You might find the code of that example useful for reference e.g. view-source:https://hikar.org/265/index.js

**Comment by fahimaltinordu** [comment]
> Hi @fahimaltinordu does this example work for you?
> 
> https://hikar.org/265/
> 
> Once it gets a GPS location, it should place two red boxes perhaps 20 metres north of your current position. You will then be able to move towards them. Best tested on a mobile device.
> 
> If that works the problem is in your code, if not the problem is with AR.js.
> Unfortunately I have little time to look into fixing this at the moment, but the link I gave you above is a good diagnostic to see whether location-based AR.js works on your device.
> 
> You might find the code of that example useful for reference i.e.: https://hikar.org/265/index.js

Hello @nickw1 i am completely stupid. Now my code works, because i currently enabled gps of safari. I think other reporters probably didnot enabled geolocation. So they see the objects on the floor and object stick with camera.

**Comment by fahimaltinordu** [comment]
https://user-images.githubusercontent.com/58973431/127686964-f2e4f2dc-7b16-4d89-b93d-997ae84d4192.MP4

**Comment by fahimaltinordu** [comment]
Hi @nickw1 sorry for disturbing,
How can i position the a-text? As you see in the video in my previous comment, the text is inside or behind the a-box, so the text looks missing. Both text and box have same lat-long. 

I used css position and zindex but didnot work. 

My aim is to show the distance of each object as label near or above the box. Is there any other way instead of using a-text same latlong with a-box?

Thank you
